src/custom/components/swap/TradePrice/index.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import styled from 'styled-components'
+import useLoadingWithTimeout from 'hooks/useLoadingWithTimeout'
+import { useIsQuoteRefreshing } from 'state/price/hooks'
+import TradePriceModComponent, { TradePriceProps } from './TradePriceMod'
+import { AnimatedImg } from 'pages/Swap'
+import loadingCowGif from 'assets/cow-swap/cow-load.gif'
+import repeat from 'assets/svg/repeat.svg'
+import { StyledBalanceMaxMini } from 'components/swap/styleds'
+
+const TradePriceMod = styled(TradePriceModComponent)`
+  ${StyledBalanceMaxMini} {
+    width: 30px;
+    height: 30px;
+    position: relative;
+    overflow: hidden;
+  }
+`
+const COW_LOADING_TIME = 4000
```

Comment:
I still think 4s is way too long.
We don't have to wait until it's done, but to me the UI indicates it's not ready until the nose-picking-cow is gone.

Reply:
maybe I'll go with @anxolin 's siuggestion from another pr then